test/unit-test/dawn/SIR/TestSIRComparison.cpp

```diff
@@ -0,0 +1,129 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/SIR/AST.h"
+#include "dawn/SIR/ASTUtil.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/Support/Casting.h"
+#include "dawn/Support/STLExtras.h"
+#include "llvm/ADT/STLExtras.h"
+#include <gtest/gtest.h>
+#include <iostream>
+
+using namespace dawn;
+
+namespace {
+
+TEST(SIRComparisonTest, Dummy) {
+  /// Works all the time, initializing the suite
+  /// checks that no exceptions are thrown with empty SIRs
+  std::unique_ptr<dawn::SIR> test01SIR = llvm::make_unique<dawn::SIR>();
+  std::unique_ptr<dawn::SIR> test02SIR = llvm::make_unique<dawn::SIR>();
+  EXPECT_TRUE(*test01SIR.get() == *test02SIR.get());
+}
+
+TEST(SIRComparisonTest, DoubleFullChain) {
+  /// Todo: this requires GTClang to be hoockable to dawn
+  /// The goal is to read a generated file with two GTClangCompilers and compare the generated SIRs
+  EXPECT_TRUE(false);
+}
+
+TEST(SIRComparisonTest, FullChainReference) {
+  /// Todo: this requires GTClang to be hoockable to dawn
+  /// The goal is to read a generated file and compare to the reference SIR generated by hand in
+  /// memory
+  EXPECT_TRUE(false);
+}
+
+TEST(SIRComparisonTest, DoubleReference) {
+  std::unique_ptr<dawn::SIR> test01SIR = llvm::make_unique<dawn::SIR>();
+  std::unique_ptr<dawn::SIR> test02SIR = llvm::make_unique<dawn::SIR>();
+
+  // Genereate a stencil
+  test01SIR->Stencils.resize(1);
+  for(auto& a : test01SIR->Stencils) {
```

Comment:
How about 
```
std::generate(test01SIR->Stencils.begin(), test01SIR->Stencils.end(), []() { return std::make_shared<sir::Stencil>(); });
```
to make it a little more condensed? ;)

Reply:
is way prettier :)